cgi-bin/DW/Console/Command/BonusIcons.pm

```diff
@@ -56,25 +58,54 @@ sub execute {
     }
 
     return $self->error( 'Invalid subcommand.' )
-        if $cmd && $cmd !~ /^(?:add|remove)$/;
+        if $cmd && $cmd !~ /^(?:add|remove|xfer)$/;
 
-    return $self->error( 'Count must be a positive integer.' )
-        unless $count =~ /^\d+$/;
-    $count += 0;
+    if ( $cmd eq 'add' || $cmd eq 'remove' ) {
 
-    if ( $cmd eq 'add' ) {
-        my $new = max( $to_u->prop( 'bonus_icons' ) + $count, 0 );
-        $to_u->set_prop( bonus_icons => $new );
-        LJ::statushistory_add( $to_u, $remote, 'bonus_icons',
-                sprintf( 'Added %d icons, new total: %d.', $count, $new ) );
-        $self->print( sprintf( 'User now has %d icons.', $new ) );
+        my $count = $cmdarg;
+
+        return $self->error( 'Count must be a positive integer.' )
+            unless $count =~ /^\d+$/ || $cmd eq 'xfer';
+        $count += 0;
+
+        if ( $cmd eq 'add' ) {
+            my $new = max( $to_u->prop( 'bonus_icons' ) + $count, 0 );
+            $to_u->set_prop( bonus_icons => $new );
+            LJ::statushistory_add( $to_u, $remote, 'bonus_icons',
+                    sprintf( 'Added %d icons, new total: %d.', $count, $new ) );
+            $self->print( sprintf( 'User now has %d icons.', $new ) );
+
+        } elsif ( $cmd eq 'remove' ) {
+            my $new = max( $to_u->prop( 'bonus_icons' ) - $count, 0 );
+            $to_u->set_prop( bonus_icons => $new );
+            LJ::statushistory_add( $to_u, $remote, 'bonus_icons',
+                    sprintf( 'Removed %d icons, new total: %d.', $count, $new ) );
+            $self->print( sprintf( 'User now has %d icons.', $new ) );
+
+        }
+
+    } elsif ( $cmd eq 'xfer' ) {
+        my $target_u = LJ::load_user( $cmdarg );
+
+        return $self->error( 'Invalid target user.' )
+            unless $target_u;
+        return $self->error( 'E-mail addresses do not match.' )
+            unless $to_u->prop( 'email' ) eq $target_u->prop( 'email' ); 
 
-    } elsif ( $cmd eq 'remove' ) {
-        my $new = max( $to_u->prop( 'bonus_icons' ) - $count, 0 );
-        $to_u->set_prop( bonus_icons => $new );
+        my $xfer_count = $to_u->prop( 'bonus_icons' );
+        my $new_to_u = max( $to_u->prop( 'bonus_icons' ) - $xfer_count, 0 );
```

Comment:
So I drew myself a pretty detailed sketch while writing this, and having tested I am pretty sure it's working as intended -- $to_u is the source account for bonus icon slots and $target_u is the sink (or receiver, or whatever). I've maintained the terminology of "$to_u" by analogy with the use of $to_u in "remove". I agree it's confusing (not... helped by the fact that GHI seems to be skipping 20 lines out of the diff, so giving completely the wrong impression of the context AFAICT) but I'm not sure how to make it clearer?